src/cmd/sanity_test/sanity_tests.go

```diff
@@ -112,6 +119,76 @@ func checkEmailRestoration(client *msgraphsdk.GraphServiceClient, testUser, fold
 				*restore.GetTotalItemCount())
 			os.Exit(1)
 		}
+
+		checkAllSubFolder(user, restore, "", messageCount)
+	}
+}
+
+func getAllSubFolder(
+	user *users.UserItemRequestBuilder,
+	r models.MailFolderable,
+	parentFolder string,
+	messageCount map[string]int32,
+) {
+	folder := user.MailFoldersById(*r.GetId())
+
+	childFolder, err := folder.ChildFolders().Get(context.Background(), nil)
+	if err != nil {
+		fmt.Printf("Error getting the drive: %v\n", err)
+		os.Exit(1)
+	}
+
+	for _, child := range childFolder.GetValue() {
+		fullFolderName := *r.GetDisplayName() + "/" + *child.GetDisplayName()
+
+		if parentFolder != "" {
+			fullFolderName = parentFolder + "/" + fullFolderName
+		}
+
+		messageCount[fullFolderName] = *child.GetTotalItemCount()
+
+		// recursively check for subfolders
+		if *child.GetChildFolderCount() > 0 {
+			parentFolder := *r.GetDisplayName()
+			getAllSubFolder(user, child, parentFolder, messageCount)
+		}
+	}
+}
+
+func checkAllSubFolder(
+	user *users.UserItemRequestBuilder,
+	r models.MailFolderable,
+	parentFolder string,
+	messageCount map[string]int32,
+) {
+	folder := user.MailFoldersById(*r.GetId())
+
+	childFolder, err := folder.ChildFolders().Get(context.Background(), nil)
+	if err != nil {
+		fmt.Printf("Error getting the drive: %v\n", err)
+		os.Exit(1)
+	}
+
+	for _, child := range childFolder.GetValue() {
+		fullFolderName := *r.GetDisplayName() + "/" + *child.GetDisplayName()
+		if parentFolder != "" {
+			fullFolderName = parentFolder + "/" + fullFolderName
+		}
+
+		if messageCount[fullFolderName] != *child.GetTotalItemCount() {
+			fmt.Println("Restore was not succesfull for: ",
+				fullFolderName,
+				"Folder count: ",
+				messageCount[fullFolderName],
+				"Restore count: ",
+				*child.GetTotalItemCount())
```

Comment:
NIt: code style standards
```suggestion
			fmt.Println(
				"Restore was not succesfull for: ", fullFolderName,
				"Folder count: ", messageCount[fullFolderName],
				"Restore count: ", *child.GetTotalItemCount())
```